CONTRIBUTING.md

```diff
@@ -0,0 +1,52 @@
+# Contributing to StackStorm pack directory
+
+You've decided to create or update a StackStorm integration pack. That's great! 
```

Comment:
The way I tried to keep it is having all "substance" (i.e. concrete examples, requirements, linting profiles, etc.) in the docs, and having a brief checklist in this document. We mostly need this file for quick reference, and the docs focus on "heavier" things like development, structure, meta, requirements, etc. This doc only contains a six-item checklist and a bunch of links to docs, and I tried to keep it as independent from the docs as possible: you'll always have to remember your meta, always have to update the version and format your code, no matter what the actual requirements for writing packs are.

So, I'd say we should merge this doc as it is (and keep it equally generic in the future), but also update st2docs to reflect some substantial things here: e.g. I don't think we have icon requirements in the docs, or a request for the version bump. Wdyt?